components/support/rust-logs/src/rust_logger.rs

```diff
@@ -0,0 +1,68 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+//! Rust Logger implementation
+//!
+//! This is responsible for taking logs from the rust log crate and forwarding them to a
+//! foreign_logger::Logger instance.
+
+use crate::foreign_logger::Logger as ForeignLogger;
+use parking_lot::RwLock;
+use std::sync::Once;
+
+// ForeignLogger to forward to
+static RUST_LOGGER: Logger = Logger::new();
+// Handles calling `log::set_logger`, which can only be called once.
+static INIT: Once = Once::new();
+
+struct Logger(RwLock<Option<Box<dyn ForeignLogger>>>);
+
+impl Logger {
+    const fn new() -> Self {
+        Self(RwLock::new(None))
+    }
+
+    fn set_foreign_logger(&self, foreign_logger: Option<Box<dyn ForeignLogger>>) {
+        *self.0.write() = foreign_logger;
+    }
+}
+
+impl log::Log for Logger {
+    fn enabled(&self, _: &log::Metadata<'_>) -> bool {
+        true
```

Comment:
I think we should implement this, but I'm worried about the performance implications of taking the lock.  I'm going to store an atomicbool instead that tracks this to avoid the overhead.